R/handler_warnings.R

```diff
@@ -0,0 +1,30 @@
+#' @title Capture Warnings
+#'
+#' @description This function captures the warnings generated by an
+#' expression without printing them to the console. It returns both the
+#' result of the expression and the captured warnings. It is useful for
+#' avoiding warnings appearing in the middle of a routine (for instance,
+#' an iterative routine).
+#'
+#' @param expr An expression to evaluate.
+#' @param warn_env An optional environment to store the warnings.
+#' If not provided, a new environment is created.
+#' @return A list containing two elements:
+#'   \item{result}{The result of evaluating the expression.}
+#'   \item{warnings}{A character vector of the warnings generated during
+#'                   the evaluation of the expression.}
+#' @keywords internal
+
+capture_warnings <- function(expr, warn_env = new.env()) {
+  warn_env$warnings <- character()
+
+  result <- withCallingHandlers(
+    expr,
+    warning = function(w) {
+      warn_env$warnings <- c(warn_env$warnings, conditionMessage(w))
+      invokeRestart("muffleWarning")
```

Comment:
Santi, please validate the use of  ` invokeRestart("muffleWarning")`  according to CRAN policy